06_puppy/madhavansu/main_test.go

```diff
@@ -0,0 +1,207 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	tassert "github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+/**
+* Map store implemented in below scenarios
+* Dynamic Map ID: MapStore implementation of Storer backed by a map
+* Static Map ID: SyncStore implementation of Storer backed by a sync.Map
+ */
+type storerImpl int
+
+var mapID uint
+
+const (
+	mem storerImpl = iota
```

Comment:
This value is meant to determine which of the two implementations are tested, so there should be two consts defined, one for `mapStore` and one for `syncStore`. The implementation is then tested via the `Storer` interface.

Then when you set up the test suite, you set up two test suites, one for each implementation.

When you make this change, you can update your tests to do all the operations once only, instead of twice as it is now.

Reply:
Updated.